os/mm/mm_heap/mm_calloc.c

```diff
@@ -81,12 +81,25 @@ FAR void *mm_calloc(FAR struct mm_heap_s *heap, size_t n, size_t elem_size)
 #endif
 {
 	FAR void *ret = NULL;
+	size_t size;
 
 	if (n > 0 && elem_size > 0) {
+		if (n > (MMSIZE_MAX / elem_size)) {
+			mdbg("Parameter n(%u) should be smaller than %u, \
+				because multiplication of n and elem_size cannot overflow the size_t.\n", n, elem_size);
+			return NULL;
+		}
+		size = n * elem_size;
+		if (size <= elem_size || size > MM_ALIGN_DOWN(MMSIZE_MAX) - SIZEOF_MM_ALLOCNODE) {
```

Comment:
```
if (n > 0 && elem_size > 0) {
    ....
    size = n * elem_size;
}
```
so, size never can be smaller than elem_size, but can be same as elem_size if n is 1
if return NULL is correct when n is 1, you'd better modify if state in line 86.

Reply:
`n` can be 1. It's not a problem. You should remove `=`. But by line 87, size can't be smaller as Taejun said.
This line is unnecessary.  
`size > MM_ALIGN....` is duplicated, it's in zalloc function.